skglm/penalties/separable.py

```diff
@@ -466,3 +466,48 @@ def is_penalized(self, n_features):
     def generalized_support(self, w):
         """Return a mask with non-zero coefficients."""
         return w != 0
+
+
+class NNLS(BasePenalty):
+    """Non-negative least square penalty."""
+
+    def __init__(self):
+        pass
+
+    def get_spec(self):
+        return ()
+
+    def params_to_dict(self):
+        return dict()
+
+    def value(self, w):
```

Comment:
I would rather put in docs ``Compute the value of the NNLS penalty at w.`` (like the other penalties)
also, I think  "positive weights" might be a bit confusing  

is there a reason for not putting the explicit implementation of the penalty val?
```python
if (w  < 0).any():
    return np.inf
else:
    return 0
```